include/swift/Runtime/Concurrency.h

```diff
@@ -891,6 +891,12 @@ ExecutorRef swift_task_getMainExecutor(void);
 SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
 bool swift_task_isCurrentExecutor(ExecutorRef executor);
 
+SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
+TypeNamePair swift_task_getExecutorRefTypeName(ExecutorRef executor);
+
+SWIFT_EXPORT_FROM(swift_Concurrency) SWIFT_CC(swift)
+ExecutorRef swift_task_getCurrentSerialExecutor();
```

Comment:
For context: I'm trying hard to not expose this as public API, and in this specific implementation we call it more than necessary, I'd like to avoid this method entirely; 

🚧 Instead we'd have a "check expected, and return the current that was NOT the expected one" API that the precondition APIs call.

Reply:
If this were to stay, this would be optional ExecutorRef or some other way to mark it